metricsaggregator/src/main/java/io/dockstore/metricsaggregator/helper/CostAggregator.java

```diff
@@ -0,0 +1,91 @@
+package io.dockstore.metricsaggregator.helper;
+
+import static io.dockstore.common.metrics.FormatCheckHelper.isValidCurrencyCode;
+import static io.dockstore.metricsaggregator.MoneyStatistics.CURRENCY;
+
+import io.dockstore.metricsaggregator.MoneyStatistics;
+import io.dockstore.openapi.client.model.Cost;
+import io.dockstore.openapi.client.model.CostMetric;
+import io.dockstore.openapi.client.model.Metrics;
+import io.dockstore.openapi.client.model.RunExecution;
+import io.dockstore.openapi.client.model.TaskExecutions;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import org.javamoney.moneta.Money;
+
+public class CostAggregator implements RunExecutionAggregator<CostMetric, Cost> {
+    @Override
+    public CostMetric getMetricFromMetrics(Metrics metrics) {
+        return metrics.getCost();
+    }
+
+    @Override
+    public Cost getMetricFromRunExecution(RunExecution runExecution) {
+        return runExecution.getCost();
+    }
+
+    @Override
+    public Optional<RunExecution> getWorkflowExecutionFromTaskExecutions(TaskExecutions taskExecutionsForOneWorkflowRun) {
+        final List<RunExecution> taskExecutions = taskExecutionsForOneWorkflowRun.getTaskExecutions();
+        if (taskExecutions != null && taskExecutions.stream().map(RunExecution::getCost).allMatch(Objects::nonNull)) {
+            // Get the overall cost by summing up the cost of each task
+            List<Cost> taskCosts = taskExecutions.stream()
+                    .map(RunExecution::getCost)
+                    .toList();
+            boolean containsMalformedCurrencies = taskCosts.stream().anyMatch(cost -> !isValidCurrencyCode(cost.getCurrency()));
```

Comment:
Not an issue  yet since we don't have examples of cost data yet. But could be follow-up issue, would it be an improvement not to skip the total cost if I understand properly if there are any malformed currencies but to provide a total of the properly formed currencies with an asterisk or warning?

Reply:
> would it be an improvement not to skip the total cost if I understand properly if there are any malformed currencies but to provide a total of the properly formed currencies with an asterisk or warning?

Hmm, in this function, I'm not sure if it would be worth it since these tasks make up one workflow execution.

> On the other hand, it might be ok if this is just ignoring one workflow run with malformed currencies and not other runs. (which I think is the case)

In this function where it aggregates a list of task executions metrics belonging to a single workflow execution into one workflow-level execution metrics, your understanding is correct. It will just ignore the one set of task executions beloning to one workflow run.

However, in the function below where we get the aggregated metric from workflow executions, we don't aggregate the metric if one of the executions is malformed. This was an existing pattern prior to this PR and also applies to the execution time metric. It sounds like we should instead ignore the one malformed workflow run and aggregate the others?

Reply:
Created a follow-up ticket https://ucsc-cgl.atlassian.net/browse/SEAB-6046